### PR TITLE
first commit, barebones functionality only

### PR DESCRIPTION
**Pull request by sashaboi**
Functionality commit , will work on more features in coming days
https://6267b17a2d50176ca5c0034e--magenta-mooncake-426b57.netlify.app/

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+  return (
+    <div className="App">
+      <h1>Card Game !</h1>
+      <p>turns done: {turnCounter}</p>
+      <div className="four-grid">
+        {flipCard.map((letter) => (
+          <div key={uuid()}>
+            <CartComponent ltr={letter.letter} />
+          </div>
```
Comment (bhavyaj12) [inline src/App.js:18]: The ltr variable can be named in a better way

File: src/App.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  const { status, flipCard, turnCounter } = UseGame();
+
+  return (
+    <div className="App">
+      <h1>Card Game !</h1>
+      <p>turns done: {turnCounter}</p>
+      <div className="four-grid">
+        {flipCard.map((letter) => (
+          <div key={uuid()}>
+            <CartComponent ltr={letter.letter} />
```
Comment (bhavyaj12) [inline src/App.js]: CartComponent? I guess you should change the name if it's wrong/picked up from E-comm

File: src/Card-Component.jsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import {BsFillQuestionCircleFill} from 'react-icons/bs';
+const CartComponent = (props) => {
+  const { flipCard, setFlipCard, setTwoCounter } = UseGame();
+  const clickHandler = () => {
+    setTwoCounter((twoCounter) => twoCounter + 1);
+    setFlipCard(
+      flipCard.map((obj) =>
+        obj.letter === props.ltr
+          ? { ...obj, tempshow: !obj.tempshow, clicked: !obj.clicked }
+          : { ...obj }
```
Comment (bhavyaj12) [inline src/Card-Component.jsx:12]: tempshow should have camel case

File: src/Card-Component.jsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+      )
+    );
+  };
+  const letterObj = flipCard.find((obj) => obj.letter === props.ltr);
+  
+
+  return (
+    <div onClick={() => clickHandler()}>
```
Comment (bhavyaj12) [inline src/Card-Component.jsx]: clickHandler can be named for the exact function it is performing, would make it more readable

File: src/game-context.js [hunk 5]
```diff
@@ -0,0 +1,84 @@
+import { useContext, createContext, useState, useEffect } from "react";
+import { dataarray } from "./dataarray";
+const gameContext = createContext();
+// dataarray.sort(() => Math.random() - 0.5);
+const GameProvider = ({ children }) => {
+  const [status, setstatus] = useState("Let's Go!");
```
Comment (bhavyaj12) [inline src/game-context.js]: setstatus should be camel case

**Review by bhavyaj12 — APPROVED** [review]
Nice game, cards aren't randomized yet though?
Nice logic implementation, have just commented on some inconsistent naming. Rest is good!

File: src/Card-Component.jsx (hunk 3 above)
Comment (bhavyaj12) [reply src/Card-Component.jsx:12]: Or can be one word like the rest in the data array

File: src/App.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import { UseGame } from "./game-context";
+import Navbar from "./Navbar";
+export default function App() {
+  const { status, flipCard, turnCounter } = UseGame();
+
+  return (
+    <div className="App">
+      <Navbar/>
+      <h1>Card Game !</h1>
+      <p>turns done: {turnCounter}</p>
```
Comment (Chiragmodi01) [inline src/App.js:13]: Your navbar contains nothing at this point. I'd suggest put these title and subtitle in the navbar comp.
I'd have only placed comps inside App.js and nothing else including 'static text' or stateful logic.

File: src/Card-Component.jsx [hunk 7]
```diff
@@ -0,0 +1,28 @@
+import { UseGame } from "./game-context";
+
+import {BsFillQuestionCircleFill} from 'react-icons/bs';
+const CartComponent = (props) => {
+  const { flipCard, setFlipCard, setTwoCounter } = UseGame();
+  const clickHandler = () => {
+    setTwoCounter((twoCounter) => twoCounter + 1);
+    setFlipCard(
+      flipCard.map((obj) =>
+        obj.letter === props.ltr
```
Comment (Chiragmodi01) [inline src/Card-Component.jsx]: destructured props and avoid using `props`.

File: src/Card-Component.jsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+          : { ...obj }
+      )
+    );
+  };
+  const letterObj = flipCard.find((obj) => obj.letter === props.ltr);
+  
+
+  return (
+    
+      <button onClick={() => clickHandler()} className={`card-component-btn `} disabled={letterObj.solved || letterObj.clicked}>
```
Comment (Chiragmodi01) [inline src/Card-Component.jsx]: Why have you put className inside braces with back ticks?

File: src/game-context.js [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+  const matchNotFoundHandler = (objsToCheck) => {
+    console.log("not matching");
+    setstatus("doesnt match :(");
+    setTurnCounter((prev) => prev + 1);
+    setTwoCounter(0);
+    setTimeout(() => {
+      setFlipCard((prev) =>
+        prev.map((obj) =>
+          obj.letter.toUpperCase() === objsToCheck[0].letter.toUpperCase() ||
+          obj.letter.toUpperCase() === objsToCheck[1].letter.toUpperCase()
```
Comment (Chiragmodi01) [inline src/game-context.js:19]: I don't seem to understand why have you use `toUpperCase` at plenty of places.
If anyways you're showing all letters in upper case in the UI, then why have you put some letter as lower case in the data?

**Review by Chiragmodi01 — COMMENTED** [review]
Other than the feedback you already got, I've added few suggestions.
And yeah, reducer would be helpful!

File: src/game-context.js (hunk 9 above)
Comment (sashaboi) [reply src/game-context.js:19]: Here, the data has two letters repeating but their case is different. The logic is sort of hacky from the array of data itself, hence the to uppercases.

File: src/App.js (hunk 1 above)
Comment (sashaboi) [reply src/App.js:18]: Changed as per suggestion.

File: src/Card-Component.jsx (hunk 3 above)
Comment (sashaboi) [reply src/Card-Component.jsx:12]: Changed as per suggestion.

File: src/game-context.js (hunk 5 above)
Comment (sashaboi) [reply src/game-context.js]: Changed as per suggestion.

File: src/App.js (hunk 6 above)
Comment (sashaboi) [reply src/App.js:13]: For now , navbar is sort of a placeholder. Will think of what to put there. will be handled in later prs.

File: src/Card-Component.jsx (hunk 7 above)
Comment (sashaboi) [reply src/Card-Component.jsx]: I agree. changed as per suggestion.

File: src/Card-Component.jsx (hunk 4 above)
Comment (sashaboi) [reply src/Card-Component.jsx]: Changed as per suggestion.

File: src/App.js (hunk 2 above)
Comment (sashaboi) [reply src/App.js]: just a happy spelling mistake, fixed.

**Pull request merged**